src/Chainweb/Pact/PactService.hs

```diff
@@ -677,6 +679,10 @@ execLocal cwtx preflight sigVerify rdepth = withDiscardedBatch $ do
     ctx <- getTxContext pm
     spv <- use psSpvSupport
 
+    for_ _psLocalMaxGasLimit $ \maxGasLimit ->
+        when (view cmdGasLimit cmd > maxGasLimit) $
+            throwM $ LocalGasLimitExceeded maxGasLimit (view cmdGasLimit cmd)
+
```

Comment:
I'm not sure which @emilypi prefers: should we reject txs with a gas limit higher than the max, or should we run the tx with the smaller of the two gas limits?

Reply:
Just a note, since @emilypi's comment is a bit outdated, the current error provides both the requested gas limit and node's limitation:

```
LocalGasLimitExceeded
    { _localMaxGasLimit :: !GasLimit
    , _localRequestedGasLimit :: !GasLimit
    }
```

Reply:
@ak3n perfect, thanks